### DISCUSSION TITLE
No updates found for a package, having a `Rejected release` log message

### DISCUSSION THREAD

**Discussion by remal** [renovatebot/renovate#31886] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

**Miniumal reproduction repository**: https://github.com/remal/renovate-reproduction-intellij-ideaIC-build-range-eap

### Context
I'm developing an IntelliJ IDEA plugin and would like to configure Renovate to update `pluginUntilBuild` property in the `gradle.properties` file.

### Chosen approach to configure Renovate
1. Configure a RegEx custom manager that processes the `gradle.properties` file.
2. Renovate is supposed to replace a value for the `pluginUntilBuild` property.
3. An updated value should be the last `XXX-EAP-SNAPSHOT` version of `com.jetbrains.intellij.idea:ideaIC` dependency in https://www.jetbrains.com/intellij-repository/snapshots/
4. `XXX-EAP-SNAPSHOT` version should be converted to `XXX` (see `extractVersion`)

### Issue
The property is not updated. An unexpected message can be found in the logs: `DEBUG: Rejected release`

What's wrong? What would you suggest me to change?

### Logs (if relevant)

Full logs: https://developer.mend.io/github/remal/renovate-reproduction-intellij-ideaIC-build-range-eap/-/job/d8b53513-063b-42cc-9d75-400b0665247c

<details><summary>Relevant logs</summary>

```

DEBUG: Looking up com.jetbrains.intellij.idea:ideaIC in repository https://www.jetbrains.com/intellij-repository/snapshots/
DEBUG: hostRules: no authentication for www.jetbrains.com
DEBUG: Using queue: host=www.jetbrains.com, concurrency=16
DEBUG: Found 38 new releases for com.jetbrains.intellij.idea:ideaIC in repository https://www.jetbrains.com/intellij-repository/snapshots/
DEBUG: GET https://www.jetbrains.com/intellij-repository/snapshots/com/jetbrains/intellij/idea/ideaIC/243.19420.21-EAP-SNAPSHOT/maven-metadata.xml = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=294)
DEBUG: Content is not found for Maven url
{
  "url": "https://www.jetbrains.com/intellij-repository/snapshots/com/jetbrains/intellij/idea/ideaIC/243.19420.21-EAP-SNAPSHOT/maven-metadata.xml"
}

DEBUG: Rejected release
{
  "datasource": "maven"
  "packageName": "com.jetbrains.intellij.idea:ideaIC"
  "registryUrl": "https://www.jetbrains.com/intellij-repository/snapshots"
  "version": "243"
}

DEBUG: PackageFiles.add() - Package file saved for base branch
{
  "baseBranch": "main"
}

DEBUG: Package releases lookups complete
{
  "baseBranch": "main"
}

DEBUG: branchifyUpgrades
DEBUG: detectSemanticCommits()
DEBUG: getCommitMessages
DEBUG: semanticCommits: detected "unknown"
DEBUG: semanticCommits: disabled
DEBUG: 0 flattened updates found: 
DEBUG: Returning 0 branch(es)
DEBUG: config.repoIsOnboarded=true
DEBUG: packageFiles with updates
{
  "baseBranch": "main"
  "config": {
    "regex": [
      {
        "deps": [
          {
            "depName": "intellij-ideaIC-build-range-eap",
            "packageName": "com.jetbrains.intellij.idea:ideaIC",
            "currentValue": "242",
            "datasource": "maven",
            "registryUrls": [
              "https://www.jetbrains.com/intellij-repository/snapshots/"
            ],
            "replaceString": "\nplatformType = IC\nplatformVersion = 2023.3.8\n\n# Supported build number ranges and IntelliJ Platform versions -> https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html\n#pluginSinceBuild = XXX - loaded automatically from `platformVersion`\npluginUntilBuild = 242.*\n\n",
            "updates": [],
            "versioning": "maven",
            "warnings": [],
            "registryUrl": "https://www.jetbrains.com/intellij-repository/snapshots",
            "packageScope": "com.jetbrains.intellij.idea",
            "currentVersion": "242"
          }
        ],
        "matchStrings": [
          "(^|\\n)\\s*platformType\\s*=\\s*IC\\s*(\\n+|$)[\\s\\S]*?(^|\\n)\\s*pluginUntilBuild\\s*=\\s*(?<currentValue>\\d+)\\.\\*\\s*(\\n+|$)",
          "(^|\\n)\\s*pluginUntilBuild\\s*=\\s*(?<currentValue>\\d+)\\.\\*\\s*(\\n+|$)[\\s\\S]*?(^|\\n)\\s*platformType\\s*=\\s*IC\\s*(\\n+|$)"
        ],
        "depNameTemplate": "intellij-ideaIC-build-range-eap",
        "packageNameTemplate": "com.jetbrains.intellij.idea:ideaIC",
        "datasourceTemplate": "maven",
        "registryUrlTemplate": "https://www.jetbrains.com/intellij-repository/snapshots/",
        "packageFile": "gradle.properties"
      }
    ]
  }
}


```

</details>

**Comment by Shegox** [comment]
I stumpled across the same. It seems the root cause is #31261, more specifically this postProcessRelease, which checks that the proposed version indeed exists on the registry: 
https://github.com/renovatebot/renovate/blob/e7543eba78fdfe3132442d2a47803e905fa5cc65/lib/modules/datasource/maven/index.ts#L259-L289

Combining this with `extractVersion` this causes Renovate to try to lookup a version which doesn't exist (because extractVersion calculates a different version).

**Comment by rarkins** [comment, marked as answer]
Converted to bug issue: https://github.com/renovatebot/renovate/issues/32266
